modules/aws/auth.go

```diff
@@ -153,3 +154,24 @@ type CredentialsError struct {
 func (err CredentialsError) Error() string {
 	return fmt.Sprintf("Error finding AWS credentials. Did you set the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables or configure an AWS profile? Underlying error: %v", err.UnderlyingErr)
 }
+
+func getNewSession(config *aws.Config) (*session.Session, error) {
+
+	if GlobalCustomEndpoints != nil {
+		config.WithS3ForcePathStyle(true).WithEndpointResolver(endpoints.ResolverFunc(endPointsCustomResolver))
+	}
+
+	return session.NewSession(config)
+}
+
+func endPointsCustomResolver(service, region string, optionalFunctions ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
+	customServiceEndpoint := (*GlobalCustomEndpoints)[service]
+	if customServiceEndpoint != "" {
```

Comment:
Continuing the thought above, if `GlobalCustomEndpoints` is just a `map` (rather than a pointer to a map), this becomes a bit safer:

```go
customServiceEndpoint, hasEndpoint := GlobalCustomEndpoints[service]
if hasEndpoint  {
  // ...
}
```

Reply:
Sure I have to verify this one, Sorry my lack of expertise in Go. This is my first try with the language